src/AzureDevOpsPolicyConfigurator/Logic/WhatIfExecuter.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using AzureDevOpsPolicyConfigurator.Data;
+using AzureDevOpsPolicyConfigurator.Interfaces;
+using Microsoft.TeamFoundation.Policy.WebApi;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+
+namespace AzureDevOpsPolicyConfigurator.Logic
+{
+    /// <summary>
+    /// WhatIfExecuter class.
+    /// </summary>
+    internal class WhatIfExecuter : PolicyExecuterBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WhatIfExecuter"/> class.
+        /// </summary>
+        /// <param name="serializer">Json serializer</param>
+        /// <param name="reader">File reader</param>
+        /// <param name="connectionProvider">Connection provider</param>
+        public WhatIfExecuter(IJsonSerializer serializer, IFileReader reader, IConnectionProvider connectionProvider)
+            : base(serializer, reader, connectionProvider)
+        {
+        }
+
+        /// <summary>
+        /// Logs the change.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="types">Types</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="repository">Git repository</param>
+        /// <param name="currentPolicy">Branch policy</param>
+        /// <param name="policy">Policy</param>
+        protected override void CreatePolicy(
+            PolicyHttpClient policyClient,
+            IEnumerable<PolicyType> types,
+            Guid projectId,
+            GitRepository repository,
+            BranchPolicies currentPolicy,
+            Policy policy)
+        {
+            Log.Info($"Policy not found in TFS, will be created. (Branch: {currentPolicy.Branch}, Repository: {repository.Name}, Type: {policy.Type})");
+            Log.Debug($"Settings is: {policy.SettingsWithScope(repository.Id)}");
+        }
+
+        /// <summary>
+        /// Logs the change.
+        /// </summary>
+        /// <param name="policyClient">Policy client</param>
+        /// <param name="types">Types</param>
+        /// <param name="projectId">Team project id</param>
+        /// <param name="repository">Git repository</param>
+        /// <param name="currentPolicy">Branch policy</param>
+        /// <param name="policy">Policy</param>
+        /// <param name="tfsPolicy">Tfs policy</param>
+        protected override void UpdatePolicy(
+            PolicyHttpClient policyClient,
+            IEnumerable<PolicyType> types,
+            Guid projectId,
+            GitRepository repository,
+            BranchPolicies currentPolicy,
+            Policy policy,
+            PolicyConfiguration tfsPolicy)
+        {
+            Log.Info($"Policy found, will be updated in TFS. (Branch: {currentPolicy.Branch}, Repository: {repository.Name}, Type: {tfsPolicy.Type.DisplayName}, TfsBranch: {tfsPolicy.GetBranch()})");
```

Comment:
Use AzureDevOps instead of TFS, as on-prem product is now also called Azure DevOps Server